storm-core/src/jvm/org/apache/storm/trident/TridentTopology.java

```diff
@@ -277,6 +277,35 @@ public Stream join(List<Stream> streams, List<Fields> joinFields, Fields outFiel
               outFields);
     }
 
+    public Stream joinSQL(Stream s1, Fields joinFields1, Stream s2, Fields joinFields2, Fields outFields) {
```

Comment:
May be you can have `join` with extra mode parameter as you mentioned. Also good to have some javadocs to explain the difference. I don't get what preserving parameters means without going through the code.